### PR TITLE
Spring boot security incorporation

### PR DESCRIPTION
**Pull request by ponomvrenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,7 +32,6 @@
             <groupId>org.springframework.boot</groupId>
```
Comment (BolsunovaNataliia) [inline pom.xml:32]: Good job! But I have a few clarifications.

File: src/main/resources/db/changelog/changes/06-create-users-roles-table.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+databaseChangeLog:
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/06-create-users-roles-table.yaml:1]: It's not correct. This table have to consist of 2 columns: user_id, role_id as one privary key

File: src/main/resources/db/changelog/changes/07-insert-data-to-users-roles-table.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+databaseChangeLog:
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/07-insert-data-to-users-roles-table.yaml]: You have to supply query instead hard code ("1" or "2") for user_id and role_id

File: src/main/resources/db/changelog/changes/07-insert-data-to-users-roles-table.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  - changeSet:
```
Comment (BolsunovaNataliia) [inline src/main/resources/db/changelog/changes/07-insert-data-to-users-roles-table.yaml]: Where is your change set to insert users to users table?

**Review by BolsunovaNataliia — APPROVED** [review]

File: src/main/java/com/example/onlinebookstore/model/User.java [hunk 5]
```diff
@@ -24,4 +36,48 @@ public class User {
     @Column(nullable = false)
     private String lastName;
     private String shippingAddress;
+    @ManyToMany
+    @JoinTable(name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id"))
+    private Set<Role> roles = new HashSet<>();
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/model/User.java:47]: Please exclude this from equals/hashCode and toString

File: src/main/resources/db/changelog/changes/06-create-users-roles-table.yaml (hunk 2 above)
Comment (andrii-hoienko) [reply src/main/resources/db/changelog/changes/06-create-users-roles-table.yaml:1]: Agree there is no need for separate column for Id, PK of this table could be composed with 2 fields

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changes/08-insert-roles-to-users-roles-table.yaml
Comment (d1sam) [inline src/main/resources/db/changelog/changes/08-insert-roles-to-users-roles-table.yaml:1]: try to improve your code and use valueComputed with preConditions for your insert.
Example:
```
      preConditions:
        - sqlCheck:
            expectedResult: 1
            sql: SELECT COUNT(*) FROM users WHERE email='admin@example.com'
        - sqlCheck:
            expectedResult: 1
            sql: SELECT COUNT(*) FROM roles WHERE name='ROLE_ADMIN'
      changes:
        - insert:
            tableName: user_role
            columns:
              - column:
                  name: user_id
                  valueComputed: (SELECT id FROM users WHERE email='admin@example.com')
              - column:
                  name: role_id
                  valueComputed: (SELECT id FROM roles WHERE name='ROLE_ADMIN')
```

File: src/main/java/com/example/onlinebookstore/repository/RoleRepository.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.example.onlinebookstore.repository;
+
+import com.example.onlinebookstore.model.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
```
Comment (d1sam) [inline src/main/java/com/example/onlinebookstore/repository/RoleRepository.java:8]:
```suggestion
@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {
```

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some changes should be done!

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**